java/ql/src/Violations of Best Practice/Exception Handling/DroppedInterruptedFlag.ql

```diff
@@ -0,0 +1,72 @@
+/**
+ * @name Missing reset of the thread interrupt flag
+ * @description Catching the `InterruptedException` without resetting the interrupt flag
+ *              may cause deadlocks.
+ * @kind problem
+ * @problem.severity recommendation
+ * @precision high
+ * @id java/unreset-interrupt-flag
+ * @tags reliability
+ *       external/cwe/cwe-391
+ *       external/cwe/cwe-833
+ */
+
+import java
+import semmle.code.java.dataflow.DataFlow
+
+/** The class `java.lang.InterruptedException`. */
+private class InterruptedException extends RefType {
+  InterruptedException() { this.hasQualifiedName("java.lang", "InterruptedException") }
+}
+
+/** The class `java.lang.Thread`. */
+private class Thread extends RefType {
+  Thread() { this.hasQualifiedName("java.lang", "Thread") }
+}
+
+/** Holds if `java.lang.InterruptedException` is caught. */
+CatchClause catchesIE(TryStmt t) {
```

Comment:
Shouldn't this QL Doc start with "Gets a XYZ"?
And the predicate should be named `getAXYZ`?